src/plugins/layout.ts

```diff
@@ -86,7 +108,15 @@ export default class Layout implements IPlugin {
   // Add node to the ShadowDom to store initial adjacent node info in a layout and obtain an index
   private discoverNode(node: Node): INodeInfo {
     let shadowNode = this.shadowDom.insertShadowNode(node, getNodeIndex(node.parentNode), getNodeIndex(node.nextSibling));
-    return shadowNode.computeInfo();
+    return this.computeInfo(shadowNode);
+  }
+
+  private computeInfo(shadowNode) {
+    let info = shadowNode.computeInfo();
+    if ((shadowNode.node as Element).tagName === "STYLE" && shadowNode.node.textContent.length === 0) {
```

Comment:
>shadowNode.node.textContent.length === 0 [](start = 60, length = 40)

Can non-empty styles contain extra rules? If yes and we are just ignoring that for perf reasons, please add a comment

Reply:
Refactored and this comment no longer applies.

---
In reply to: [243077973](https://github.com/Microsoft/clarity-js/pull/116#discussion_r243077973) [](ancestors = 243077973)